DS_Session4/src/charaterSequence/StringCounting.java

```diff
@@ -0,0 +1,74 @@
+package charaterSequence;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+/**
+ * String Counting class is created to calculate the unique character in a string .
+ *
+ * @author Aman Gautam
+ * Dated- 3/8/2019
+ *
+ */
+public class StringCounting {
+	
+	Map<String,Integer> map = new HashMap<String,Integer>(); 
+	Set<String> set  ;
+	
+	/**
+	 * calculation method stores the unique character to inside the map no of unique charactesrs.
+	 * 
+	 * 
+	 * @param inputString
+	 * @return no of unique character
+	 */
+	public int calculation(String inputString){
+		set = new HashSet<String>(); 
+		int answer=0;  
+		for(int count_i=0;count_i<inputString.length();count_i++){
+			if(!" ".equals(Character.toString(inputString.charAt(count_i)))){
+				set.add(Character.toString(inputString.charAt(count_i))); 
+			}
+		}
+		answer = set.size();
+		return answer;
+	}
+	
+	/**
+	 *  this method verifies that if the string is already present in the 
+	 *  cache memory .
+	 *  
+	 * @param inputString
+	 * @return cacheValue
+	 */
+	public int cache(String inputString){
```

Comment:
If a method is not being used outside this file, then change it's access to private.